staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -349,6 +358,15 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	errorStream, err := pf.streamConn.CreateStream(headers)
 	if err != nil {
 		runtime.HandleError(fmt.Errorf("error creating error stream for port %d -> %d: %v", port.Local, port.Remote, err))
+		// when a requests times out creating a stream (it usually is one of
+		// subsequent requsts) it means that the remote channel is broken and
+		// we should close port-forward
+		if errors.Is(err, spdystream.ErrTimeout) {
```

Comment:
it seems safer to always tear down overall when initialization fails for any reason ... I'm really unsure of this specific error detection

Reply:
Updated.